lib/puppet/configurer.rb

```diff
@@ -126,6 +126,73 @@ def convert_catalog(result, duration, facts, options = {})
     catalog
   end
 
+  def warn_name_length(name, length, max_length)
+    Puppet.warning _("Fact %{name} with length: '%{length}' exceedes the length limit: %{limit}") % { name: name, length: length, limit: max_length }
+  end
+
+  def warn_size_length(size, max_size)
+    Puppet.warning _("The current fact size: %{size} exceeds the number of top facts limit: %{max_size}") % { size: size, max_size: max_size }
+  end
+
+  def warn_value_length(key, value, length, max_length)
+    Puppet.warning _("Fact '%{key}' with value '%{value}' and the value length: '%{length}' exceeds the value length limit: %{max_length}") % { key: key, value: value, length:length, max_length: max_length }
+  end
+
+  def check_fact_name_length(name)
+    max_length = Puppet[:max_factnamelength]
+    return if max_length == -1
+
+    max_length = 100 if max_length == 0
```

Comment:
Since we are checking `bytesize` i think these limits need to be adjusted. I would suggest to move them in some contestants at the beginning of this file and add a descriptive comment to them, eg:
```ruby
# max fact name length in bytes
MAX_FACT_NAME_LENGTH = 100

```